aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -181,14 +169,16 @@ Tensor logdet(const Tensor& self) {
   TORCH_CHECK((at::isFloatingType(self.scalar_type()) || at::isComplexType(self.scalar_type())),
               "Expected a floating point tensor as input");
 
-  c10::ExclusivelyOwned<Tensor> det_P, diag_U;
-  std::tie(det_P, diag_U) = _lu_det_P_diag_U(self);
-  Tensor det_sign = diag_U->sign().prod(-1).mul_(*det_P);
+  Tensor pivs, lu;
+  std::tie(lu, pivs, std::ignore) = at::linalg_lu_factor_ex(self);
+  const auto det_P = _lu_det_P(lu, pivs);
+  const auto diag_U = lu.diagonal(0, -2 ,-1);
```

Comment:
I don't see a move from det_P or diag_U in this function. where is it?

Reply:
Sorry, what do you mean?

Reply:
this used to use c10::ExclusivelyOwned for det_P and diag_U. The original comment said that that wasn't necessary because we could move out of those tensors, but I don't see moves from those tensors in this function.

Reply:
ExclusivelyOwned saved the atomic reference count decrement to destroy the returned tensors in the callers of _lu_det_P_diag_U. I agree that it didn't do anything for lu_det_P_diag_U itself.

Reply:
Ahhh, right. Subtle. Thank you for the explanation!